macros/compat/src/main/scala-2.11/enumeratum/ContextUtils.scala

```diff
@@ -37,4 +37,10 @@ object ContextUtils {
   def constructorName(c: Context): c.universe.TermName = {
     c.universe.termNames.CONSTRUCTOR
   }
+
+  def evalTree[ValueType](c: Context)(expr: c.universe.Tree): ValueType = {
+    @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements")) // false positive bug in WartRemover?
+    val evaluated = c.eval(c.Expr[ValueType](c.untypecheck(c.typecheck(expr))))
```

Comment:
`eval` will only evaluate untyped expressions. For some reason it says `expr` is typechecked and refuses to work. Performing an `untypecheck` will complain that `expr` is actually not typed so it can't untype it.... So we run `typecheck` to make it very typed and then `untypecheck` it so it's not typed and `eval` likes it.

I wasted days on this...